### PR TITLE
8 carrier label helper

### PR DESCRIPTION
**Pull request by bealdav**

### CODE REVIEW

**Comment by alexis-via** [comment]
:+1:

File: base_delivery_carrier_label/stock.py [hunk 1]
```diff
@@ -45,6 +46,45 @@ def _complete_name(self, name, args):
...
+            operations = pack_op_obj.search(
+                ['|',
+                 '&',
+                 ('package_id', '=', package.id),
+                 ('result_package_id', '=', False),
+                 ('result_package_id', '=', package.id),
+                 ('product_id', '!=', False),
+                 ])
+            for operation in operations:
+                weight += operation.product_id.weight * operation.product_qty
```
Comment (alexis-via) [inline base_delivery_carrier_label/stock.py]: Just one remark : if the category of the unit of measure is weight, we should compute the weight differently...

File: base_delivery_carrier_label/stock.py [hunk 2]
```diff
@@ -279,6 +319,33 @@ def _get_label_sender_address(self):
...
+            if package:
+                weight = package.get_weight()
+                package.write({'weight': weight})
+        return
+
+    @api.multi
+    def _check_existing_shipping_label(self):
+        """ Check that labels don't already exist for this picking """
+        self.ensure_one()
+        labels = self.pool['shipping.label'].search([
```
Comment (yvaucher) [inline base_delivery_carrier_label/stock.py]: s/pool/env

File: base_delivery_carrier_label/stock.py [hunk 3]
```diff
@@ -279,6 +319,33 @@ def _get_label_sender_address(self):
...
+            ('res_id', '=', self.id),
+            ('res_model', '=', 'stock.picking')])
+        if labels:
+            raise UserError(
+                _('Error:'),
```
Comment (yvaucher) [inline base_delivery_carrier_label/stock.py]: You should remove this line.

File: base_delivery_carrier_label/stock.py [hunk 4]
```diff
@@ -45,6 +46,45 @@ def _complete_name(self, name, args):
                 res[pack.id] += ' %s kg' % pack.weight
         return res
 
+    @api.multi
+    def get_weight(self):
```
Comment (yvaucher) [inline base_delivery_carrier_label/stock.py:98]: Would be great to have a test that cover this method

**Comment by bealdav** [comment]
@alexis-via @guewen @yvaucher 
please look at https://github.com/OCA/carrier-delivery/wiki/base_delivery_carrier_label-evolution
Are you ok ?

**Comment by bealdav** [comment]
ping @yvaucher @guewen tests in base delivery label to review from @hparfr. Thanks

**Comment by guewen** [comment]
Great! Thanks :+1:

**Comment by yvaucher** [comment]
Thanks :+1:

**Pull request merged**